lib/utils/slices/slices.go

```diff
@@ -18,21 +18,24 @@
 
 package slices
 
-import (
-	"cmp"
-	"slices"
-)
-
 // FilterMapUnique applies a function to all elements of a slice and collects them.
 // The function returns the value to collect and whether the current element should be included.
 // Returned values are sorted and deduplicated.
-func FilterMapUnique[T any, S cmp.Ordered](ts []T, fn func(T) (s S, include bool)) []S {
-	ss := make([]S, 0, len(ts))
+func FilterMapUnique[T any, S any](ts []T, fn func(T) (s S, include bool)) []S {
+	ss := make([]S, 0)
+	seen := make(map[key[S]]struct{})
 	for _, t := range ts {
 		if s, include := fn(t); include {
-			ss = append(ss, s)
+			if _, ok := seen[key[S]{val: s}]; !ok {
+				seen[key[S]{val: s}] = struct{}{}
```

Comment:
Maps will compile with an interface type as the key type even if the concrete value is not comparable, specifically to support things that are conditionally not comparable, but just because it compiles it doesn't mean that it'll work correctly. This will panic at runtime if used on a type that's not comparable, hacky `key[S]` thing notwithstanding.

`S` should have a constraint of `comparable` rather than `any`, and in that case we can forgo the `key` type and use a `map[S]struct{}` directly.

Reply:
Thank you for the insight, it was really helpful.
Can you please take another look?